src/signer/op-signer/launcher.star

```diff
@@ -0,0 +1,272 @@
+        config=config,
+    )
+
+    service = plan.add_service(
+        name=params.service_name,
+        config=config,
+    )
+
+    return struct(service=service, credentials=credentials)
+
+
+def get_service_config(plan, params, credentials, private_keys, config):
+    return ServiceConfig(
+        image=params.image,
+        labels=params.labels,
+        ports=_net.ports_to_port_specs(params.ports),
+        cmd=[],
+        files={
+            _SIGNER_CREDENTIALS_DIR: credentials.artifact,
+            _SIGNER_CONFIG_DIR: config.artifact,
+            _SIGNER_PRIVATE_KEYS_DIR: Directory(
+                artifact_names=[
+                    private_key.artifact for private_key in private_keys.values()
+                ]
+            ),
+        },
+        env_vars={
+            "OP_SIGNER_TLS_CA": "{}/{}".format(
+                _SIGNER_CREDENTIALS_DIR, credentials.ca.crt
+            ),
+            "OP_SIGNER_TLS_CERT": "{}/{}".format(
+                _SIGNER_CREDENTIALS_DIR,
+                credentials.hosts[params.service_name].tls.crt,
+            ),
+            "OP_SIGNER_TLS_KEY": "{}/{}".format(
+                _SIGNER_CREDENTIALS_DIR,
+                credentials.hosts[params.service_name].tls.key,
+            ),
+            "OP_SIGNER_RPC_PORT": str(params.ports[_net.HTTP_PORT_NAME].number),
+            "OP_SIGNER_SERVICE_CONFIG": "{}/{}".format(
+                _SIGNER_CONFIG_DIR, config.config
+            ),
+        },
+        private_ip_address_placeholder=_ethereum_package_constants.PRIVATE_IP_ADDRESS_PLACEHOLDER,
+    )
+
+
+# This function generates the credentials to be used for communication between the signer and the clients
+def _create_credentials(plan, params, hosts, script_artifacts, registry):
+    generate_credentials = plan.run_sh(
+        name="{}-gen-creds".format(params.service_name),
+        description="Generate local credentials for op-signer",
+        run="/scripts/gen-local-creds.sh all {}".format(" ".join(hosts)),
+        image=registry.get(_registry.OPENSSL),
+        files={"/scripts": script_artifacts},
+        env_vars={
+            "TLS_DIR": "/tls",
+        },
+        store=[
+            StoreSpec(
+                src="/tls",
+                name="{}--creds".format(params.service_name),
+            ),
+        ],
+    )
+
+    return struct(
+        # We'll return the credentials artifact itself
+        artifact=generate_credentials.files_artifacts[0],
+        # As well as the relative paths to all the files within it
+        ca=struct(crt="ca.crt", key="ca.key"),
+        hosts={
+            hostname: struct(
+                tls=struct(
+                    crt="{}/tls.crt".format(hostname),
+                    key="{}/tls.key".format(hostname),
+                )
+            )
+            for hostname in hosts
+        },
+    )
+
+
+def _create_signer_config(
+    plan, params, network_params, clients, private_keys, script_artifacts
```

Comment:
`script_artifacts` arg is unused. Can it be removed?

Reply:
Done